js/src/ReactView.js

```diff
@@ -0,0 +1,207 @@
+                };
+            }
+        } else if (model.keys().includes('value')) {
+            /* This widget uses value as main property which should be changed in the onChange callback. The new value
+             * is, depending on the widget, stored in the event or value argument of the onChange callback.
+             */
+            childProps.onChange = (e, value) => {
+                if (model instanceof ToggleButtonGroupModel || model instanceof TabsModel) {
+                    model.set('value', value);
+                } else if (e.target && e.target.value !== undefined) {
+                    model.set('value', e.target.value);
+                } else if (value !== undefined) {
+                    model.set('value', value);
+                }
+                model.save_changes(model.callbacks(view));
+            };
+        }
+
+        (model.get('_events') || [])
+            .forEach((eventStr) => {
+                const [event, ...modifiers] = eventStr.split('|').map(s => s.trim());
+                const existingFn = childProps[event];
+                childProps[event] = (e, ...args) => {
+                    if (modifiers.includes('preventDefault')) {
+                        e.preventDefault();
+                    }
+                    if (existingFn) {
+                        existingFn(e, ...args);
+                    }
+                    model.send({ event, data: args }, model.callbacks(view));
+                };
+            });
+        childProps.key = model.cid;
+        childProps.cid = view.cid + model.cid;
+
+        return childProps;
+    }
+
+    convertModels(value, ancestors) {
+        if (value instanceof ReactWidgetModel) {
+            return React.createElement(value.getReactComponent(), this.makeProps(value, ancestors));
+        }
+        const { view } = this.props;
+        if (value instanceof WidgetModel) {
+            return React.createElement(WidgetComponent, {
+                model: value,
+                key: value.cid,
+                view,
+            });
+        }
+        if (Array.isArray(value)) {
+            return value.map(e => this.convertModels(e, ancestors));
+        }
+        return value;
+    }
+
+    listModels(value) {
+        if (value instanceof ReactWidgetModel) {
+            const childModels = _.flatten(value.values().map(v => this.listModels(v)))
+                .filter(_.identity);
+            return [value].concat(childModels);
+        }
+        if (Array.isArray(value)) {
+            return _.flatten(value.map(v => this.listModels(v))).filter(_.identity);
+        }
+        return false;
+    }
+
+    render() {
+        /* When any property of any backbone model this model hierarchy changes, re render the whole React virtual DOM.
+         * First unregister all change callbacks from models previously rendered, then add change callbacks to all
+         * models currently in the model hierarchy.
+         */
```

Comment:
Add sth like:
Props (which ancestors can read) cannot change, however, the backbone model *can* change. Therefore we have to rerender.

Reply:
I will improve this comment later, the release should not have to wait on this.